### ISSUE TITLE
Untitled story 6132

### ISSUE THREAD

**Issue by diegoliberman** [instedd/cdx#1390]

**Comment by diegoliberman** [comment]
In my search for the next issue for Johannes, I arrived here.

I think that while DiegoD moves forward with 2 (#1388) & 3 (#1287), Johannes could start this one.  I think in this issue we could do everything, except the creation of the log when the sample is transferred.  It would be more convenient to wait for rebasing with 2 & 3 before implementing that.  

We could create the database model, and we could manually create records in order to test the UI.  

@devduarte @ysbaddaden @straight-shoota what do you think?

**Comment by straight-shoota** [comment]
*The Confirm Receipt action could show the date in a tooltip.*: What date is there supposed to be shown there? When the `Confirm Receipt` action is open, there's no confirmation date.

**Comment by diegoliberman** [comment]
> _The Confirm Receipt action could show the date in a tooltip._: What date is there supposed to be shown there? When the `Confirm Receipt` action is open, there's no confirmation date.

@gbatiston @jkicillof I think it should be the date that the Sample was transferred.  Right?

**Comment by diegoliberman** [comment]
@gbatiston bear in mind that the `Confirm Receipt` action is shown only to the recipient.  Hence, the POV is of the Recipient, not the Sender.   Use this [design](https://projects.invisionapp.com/d/main#/console/4103589/462385786/preview) as a reference.

Showing the date that it was sent sounds good to me. @straight-shoota do you think that's possible?  I'd show something like `Sent on <>`

**Comment by straight-shoota** [comment]
Definitely. I thought about that as well, actually. The date the transfer is started is already recorded, so it's just about showing it.
The question is just how to present that data.

This was rather unspecified, so I just picked something for #1491. But I'm not happy with the wording.

Currently it's this:

* Sender
  * In transit: `Pending`
  * Confirmed: `Sent on %{confirmed_at}`
* Receiver
  * In transit: `Confirm receipt` (action)
  * Confirmed: `Received on %{confirmed_at}`

The semantics of `Sent`  is problematic: It's not clear whether it refers to the local part of the process ("I have sent that sample away") or the overall transfer ("The sample has been sent to the receiver (and was confirmed)"). The current implementation as noted here uses the latter semantics, while the proposal to use `Sent on %{created_at}` refers to the former.

Another potential issue with showing the start date is on the receiver side: If the cell is supposed to contain both the date and an action button with a label, that makes it quite long: `Sent on %{created_at}. Confirm receipt`

![grafik](https://user-images.githubusercontent.com/466378/156451560-8d64162f-a733-4749-9823-3e568a559c2f.png)

Maybe it's okay. On smaller viewports, the latter part might be cut off. But that's just part of the button label. As long as the icon is still visible, we're good. Perhaps it might be an option to completely hide the label on small vieports, and show only the icon for the confirm action?

On a related note: The transfer start date might also be a good candidate for a filter. For user stories like: I want to see all the samples that have been sent more than a week ago and are not yet confirmed (makes sense for both sender and receiver).

**Comment by diegoliberman** [comment]
Regarding the semantics, I can see that it might be confusing.  Not sure how to improve it, though.  

Just to help myself think about this, I mention that there are 2 separated actions:
1. `I sent the sample`
2. `I confirmed the receipt of sample`

What if we go like this:

* Sender
  * In transit: `Sent on %{created_at}`
  * Confirmed: `Delivery confirmed on %{confirmed_at}`
* Receiver
  * In transit: `Confirm receipt` (action). --> As you mentioned, adding the date + action will use too much space.  In fact, it was originally designed like that, and we decided to leave the action only, and show a tooltip on hover with the date info (`Received on %{created_at}`)
  * Confirmed: `Receipt confirmed on %{confirmed_at}`

@gbatiston these are just suggestions, please feel free to help us define this messaging.

**Comment by gbatiston** [comment]
> @gbatiston bear in mind that the `Confirm Receipt` action is shown only to the recipient. Hence, the POV is of the Recipient, not the Sender. Use this [design](https://projects.invisionapp.com/d/main#/console/4103589/462385786/preview) as a reference.
> 
> Showing the date that it was sent sounds good to me. @straight-shoota do you think that's possible? I'd show something like `Sent on <>`

@diegoliberman  I must have been confused by the fact that the ticket discloses everything that happens from the Sender's perspective, and its log activity.

**Comment by gbatiston** [comment]
@straight-shoota I like your suggestions, I'd only say that for the Receiver-Confirmed case, I'd say `Reception confirmed` more than `Receipt`. But that's probably me just bike-shedding :)))

**Comment by straight-shoota** [comment]
I've updated https://github.com/instedd/cdx/pull/1491 as proposed.

Also added a tooltip with detailed information on every status cell.

**Comment by straight-shoota** [comment]
I'm also wondering about sort order of the transfer list. Currently, it's undefined. A good starting point would be to sort by date of creating descendingly. So you always see the newest ones on top.

However, it might also make sense to show newly confirmed ones up. That means sorting by the date of the latest update (created or confirmed).
I believe that would probably be best. WDYT @gbatiston?

**Comment by diegoliberman** [comment]
We decided that the sort should be by creation date descending.  At least for the time being.

**Comment by straight-shoota** [comment]
For documentation: These are the status displays we ended up with:

![grafik](https://user-images.githubusercontent.com/466378/158197376-36969f2e-3e73-451c-8e82-a212a86dbe47.png)
![grafik](https://user-images.githubusercontent.com/466378/158197999-d9a2b228-b245-4574-9ac6-c819d0c58393.png)

All have a tool tip with more information.

The latter one (missing authorization to confirm) could be improved with a help message (#1532).

@jkicillof I suppose this should be updated in the designs?

**Comment by diegoliberman** [comment]
Verified in `Version: dev-525a860 (build 1058)`
